plugin/src/main/java/org/gradle/testretry/TestRetryTaskExtension.java

```diff
@@ -81,63 +81,107 @@ public interface TestRetryTaskExtension {
-         *
+         * <p>
          * If no patterns are specified, all classes (that also meet other configured filters) will be included.
          */
         SetProperty<String> getIncludeClasses();
 
         /**
          * The patterns used to include tests based on their class level annotations.
-         *
+         * <p>
          * The pattern string matches against the qualified class names of a test class's annotations.
          * It may contain '*' characters, which match zero or more of any character.
-         *
+         * <p>
          * A class need only have one annotation matching any of the patterns to be included.
-         *
+         * <p>
          * Annotations present on super classes that are {@code @Inherited} are considered when inspecting subclasses.
-         *
+         * <p>
          * If no patterns are specified, all classes (that also meet other configured filters) will be included.
          */
         SetProperty<String> getIncludeAnnotationClasses();
 
         /**
          * The patterns used to exclude tests based on their class name.
-         *
+         * <p>
          * The pattern string matches against qualified class names.
          * It may contain '*' characters, which match zero or more of any character.
-         *
+         * <p>
          * A class name only has to match one pattern to be excluded.
-         *
+         * <p>
          * If no patterns are specified, all classes (that also meet other configured filters) will be included.
          */
         SetProperty<String> getExcludeClasses();
 
         /**
          * The patterns used to exclude tests based on their class level annotations.
-         *
+         * <p>
          * The pattern string matches against the qualified class names of a test class's annotations.
          * It may contain '*' characters, which match zero or more of any character.
-         *
+         * <p>
          * A class need only have one annotation matching any of the patterns to be excluded.
-         *
+         * <p>
          * Annotations present on super classes that are {@code @Inherited} are considered when inspecting subclasses.
-         *
+         * <p>
          * If no patterns are specified, all classes (that also meet other configured filters) will be included.
          */
         SetProperty<String> getExcludeAnnotationClasses();
 
     }
 
+
+    /**
+     * The set of criteria specifying which test classes must be retried as a whole unit.
+     */
+    ClassRetryCriteria getClassRetry();
+
+    /**
+     * The set of criteria specifying which test classes must be retried as a whole unit.
+     */
+    void classRetry(Action<? super ClassRetryCriteria> action);
+
+    /**
+     * A set of criteria specifying which test classes must be retried as a whole unit.
+     */
+    interface ClassRetryCriteria {
+
+        /**
+         * The patterns used to include tests based on their class name.
```

Comment:
This JavaDoc has to be adjusted.

We should probably also make clear that those criteria are only used for tests that pass the filter? You cannot add additional test classes for retry here.

Reply:
I agree that this should be clarified but would do so in the class-level Javadoc.

Reply:
Well, the pattern includes these classes to be retried as a whole, so I don't think it is wrong.

> I agree that this should be clarified but would do so in the class-level Javadoc.

I don't quite understand the interface and the getter/action method are documented accordingly.

Do you have any suggestions to improve this?

Reply:
I added a proposal above.